secant/Util/RecoveryPhraseStorage.swift

```diff
@@ -0,0 +1,232 @@
+            /// The data in the keychain item can be accessed only while the device is unlocked by the user.
+            /// This is recommended for items that need to be accessible only while the application is in the foreground.
+            /// Items with this attribute do not migrate to a new device.
+            /// Thus, after restoring from a backup of a different device, these items will not be present.
+            kSecAttrAccessible as String: kSecAttrAccessibleWhenUnlockedThisDeviceOnly
+        ]
+        
+        return query
+    }
+    
+    static func restoreQuery(forAccount account: String = "", andKey forKey: String) -> [String: Any] {
+        var query = baseQuery(forAccount: account, andKey: forKey)
+        query[kSecMatchLimit as String] = kSecMatchLimitOne
+        query[kSecReturnData as String] = kCFBooleanTrue
+        query[kSecReturnRef as String] = kCFBooleanFalse
+        query[kSecReturnPersistentRef as String] = kCFBooleanFalse
+        query[kSecReturnAttributes as String] = kCFBooleanFalse
+        
+        return query
+    }
+}
+
+private extension RecoveryPhraseStorage {
+    // MARK: - Recovery Phrase Interface
+    
+    /// Use carefully:  Deletes seed phrase from the keychain!!!
+    @discardableResult
+    func deleteRecoveryPhrase() -> Bool {
+        deleteData(forKey: Constants.zcashRecoveryPhrase)
+    }
+
+    func recoveryPhrase() throws -> String {
+        guard let data = data(forKey: Constants.zcashRecoveryPhrase) else {
+            throw KeychainError.noDataFound
+        }
+        
+        return String(decoding: data, as: UTF8.self)
+    }
+
+    func setRecoveryPhrase(_ phrase: String) throws {
+        guard let data = phrase.data(using: .utf8) else {
+            throw KeychainError.encoding
+        }
+            
+        try setData(data, forKey: Constants.zcashRecoveryPhrase)
+    }
+
+    // MARK: - Birthday Interface
+    
+    func birthday() throws -> BlockHeight {
+        guard let data = data(forKey: Constants.zcashBirthday) else {
+            throw KeychainError.noDataFound
+        }
+
+        do {
+            if let birthday = try NSKeyedUnarchiver.unarchivedObject(ofClass: NSNumber.self, from: data) {
+                return birthday.intValue
+            }
+            
+            throw KeychainError.decoding
+        } catch {
+            throw error
+        }
+    }
+
+    /// Use carefully:  Deletes birthday from the keychain!!!
+    @discardableResult
+    func deleteBirthday() -> Bool {
+        deleteData(forKey: Constants.zcashBirthday)
+    }
+
+    func setBirthday(_ birthday: BlockHeight) throws {
+        do {
+            let data = try NSKeyedArchiver.archivedData(withRootObject: NSNumber(value: birthday), requiringSecureCoding: true)
+            
+            try setData(data, forKey: Constants.zcashBirthday)
+        } catch {
+            throw error
+        }
+    }
+    
+    // MARK: - General Interface
```

Comment:
Is this an interface, or is this comment not really needed?

Reply:
removed